include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h

```diff
@@ -609,16 +609,23 @@ struct __scan
         __init_processing<_Tp> __use_init{};
 
         ::std::size_t __shift = 0;
-        __internal::__invoke_if_not(_Inclusive{}, [&]() { __shift = 1; });
+        __internal::__invoke_if_not(_Inclusive{}, [&__shift]() { __shift = 1; });
```

Comment:
My proposal: const auto __shift = _Inclusive().value ? 0 : 1;

Reply:
_Inclusive() - is a kind of compile time information. In general we have to use the compile information in the compile time.

Reply:
Sure, it's compile-time const. But they also has value field in run-time. And it's make ability to declare it as const and initalize in one line.

Reply:
Rather, so:
`constexpr ::std::size_t __shift = _Inclusive().value ? 0 : 1;`
It is compile time expression and look more readable.